numpy/lib/function_base.py

```diff
@@ -1904,10 +1925,76 @@ def diff(a, n=1, axis=-1):
     slice1 = tuple(slice1)
     slice2 = tuple(slice2)
     if n > 1:
-        return diff(a[slice1]-a[slice2], n-1, axis=axis)
-    else:
+        return diff(a[slice1]-a[slice2], n-1, axis=axis, 
+                    prepend=prepend, append=append)
+
+    # if nothing to add to either side, fast track the result
+    elif prepend is None and append is None:
         return a[slice1]-a[slice2]
 
+    else:
+        # logic to handle append input
+        # l_end is the length of padding at the end
+        if append is None:
+            l_end = 0
+        else:
+            append = np.asanyarray(append)
+            if append.ndim == 0:
+                l_end = 1
+            elif append.ndim != a.ndim:
+                raise ValueError('prepend must have the same dimension as ' +\
+                                 'the input array')
+            else:
+                l_end = append.shape[axis]
+
+        # logic to handle prepend input
+        # l_begin is the length of padding at the beginning
+        if prepend is None:
+            l_begin = 0
+        elif isinstance(prepend, str) and prepend == 'first':
+            prepend = a.take([0], axis)
```

Comment:
I think it should return an empty list - because that way, `cumsum` is still the inverse.